addons/account_facturx/models/account_move.py

```diff
@@ -239,6 +239,24 @@ def find_partner(partner_type):
                                 invoice_line_form.price_unit = float(line_elements[0].text) / float(quantity_elements[0].text)
                             else:
                                 invoice_line_form.price_unit = float(line_elements[0].text)
+                            # For Gross price, we need to check if a discount must be taken into account
+                            line_elements = element.xpath('.//ram:AppliedTradeAllowanceCharge',
+                                                          namespaces=tree.nsmap)
+                            if line_elements:
+                                line_elements = element.xpath(
+                                    './/ram:AppliedTradeAllowanceCharge/ram:CalculationPercent', namespaces=tree.nsmap)
+                                if line_elements:
+                                    invoice_line_form.discount = float(line_elements[0].text)
+                                else:
+                                    # if discount not available, it will be computed from the gross and net prices.
+                                    line_elements = element.xpath('.//ram:NetPriceProductTradePrice/ram:ChargeAmount',
+                                                                  namespaces=tree.nsmap)
+                                    if line_elements:
+                                        quantity_elements = element.xpath(
+                                            './/ram:NetPriceProductTradePrice/ram:BasisQuantity', namespaces=tree.nsmap)
```

Comment:
I don't know Factur-X at all but according to the code I'm just wondering if the tag `<ram:BasisQuantity>` is only present if the value of `<ram:ChargeAmount>` is the total charged for that product instead of the unit price. Regarding `<ram:BilledQuantity>`,I think that this tag will always be present.

Reply:
BasisQuantity is 

> The number of item units to which the price applies.

> Management of the Item price base quantity (BT-149) at the line level. This value gives the number
of units on which the price applies (for example if the value is 3, this means that the unit price is for
3 boxes).

While the BilledQuantity is 

> The quantity of items goods or services) that is charged in the Invoice line.

> the Invoice line net amount is equal to the Item net price divided by the Item price 
base quantity unit of measure code multiplied by the Invoiced quantity, rounded to 2 
digits, minus the sum of line allowances